### ISSUE TITLE
Change interpolation algorithm

### ISSUE THREAD

**Issue by berland** [equinor/pyscal#44]
The interpolation code handles saturation endpoints naively. 

The current code illustrates how bad it will look in an extreme setting:
```python
from pyscal import *
import matplotlib.pyplot as plt 
import copy
wo0 = WaterOil(swl=0.00)
wo1 = WaterOil(swl=0.5)
wo0.add_corey_water(nw=1)
wo1.add_corey_water(nw=1)
wo0.add_corey_oil(now=3)
wo1.add_corey_oil(now=3)
ip = WaterOil(swl=0.05)
utils.interpolator(ip, copy.deepcopy(wo0), copy.deepcopy(wo1), 0.5)
(fig, ax) = plt.subplots()
wo0.plotkrwkrow(ax=ax, color='blue')
wo1.plotkrwkrow(ax=ax, color='red')
ip.plotkrwkrow(ax=ax, color='green')
plt.show()
```
The green curves are here the interpolants between red and blue, with an unwanted discontinuity at sw=0.5.

![Screenshot from 2019-10-24 22-35-50](https://user-images.githubusercontent.com/306834/67523364-d2a0b280-f6ae-11e9-8938-bcd3f3600dab.png)

**Comment by berland** [comment]
This particular example will be fixed by normalizing Sw between swl and 1, and interpolate in the normalized saturation domain. Swl will be separately interpolated linearly, and the interpolated permeability will be applied on the denormalized saturation values.

**Comment by vkip** [comment]
Suggestion: 
- Interpolate horizontal endpoints --> [Swa, Swb]
- Create new table on [Swa, Swb] by interpolating high/low mapped to this interval.

**Comment by berland** [comment]
Also solve #42 here.

**Comment by berland** [comment]
Another example for `GasOil` 
```python
from pyscal import *
import matplotlib.pyplot as plt 
import copy
go0 = GasOil(sorg=0.5)
go1 = GasOil(sorg=0.2)
go0.add_corey_gas(ng=2, krgend=0.9)
go1.add_corey_gas(ng=2, krgend=0.8)
go0.add_corey_oil(nog=2)
go1.add_corey_oil(nog=2)
ip = GasOil()
utils.interpolator(ip, copy.deepcopy(go0), copy.deepcopy(go1), 0.5, sat='sg', kr1='krg', kr2='krog')
(fig, ax) = plt.subplots()
go0.plotkrgkrog(ax=ax, color='blue')
go1.plotkrgkrog(ax=ax, color='red')
ip.plotkrgkrog(ax=ax, color='green')
plt.show()
```
![image](https://user-images.githubusercontent.com/306834/67563439-f8669f80-f720-11e9-98cd-7c5dae641337.png)

The green kink at `sg=0.5` is not easily avoided, as knowledge of `sorg` is not easily deduced from the tabular data with `sg` and `krg` (Some programming would be able to split the curves into non-linear and linear parts, and do interpolation separately on each domain).

**Comment by berland** [comment]
#51  has been merged, and is critical in order to solve this issue.

**Comment by berland** [comment]
wip: #52

**Comment by berland** [comment]
Solved in #52
